lib/opencv_nx.ex

```diff
@@ -7,82 +7,74 @@ defmodule OpenCV.Nx do
   list.
   """
 
+  @compile {:no_warn_undefined, Nx}
+
   @doc """
-  Transform an OpenCV.Mat `#Reference` to `#Nx.tensor`
+  Transform an OpenCV.Mat `#Reference` to `Nx.tensor`.
+
+  The resulting tensor is in the shape `{width, height, channels}`.
+
   ### Example
+
   ```elixir
-  {:ok, mat} = OpenCV.imread("/path/to/exist/img.png")
-  nx_tensor = OpenCV.Nx.to_nx(mat)
-   #Nx.Tensor<
-      u8[1080][1920][3]
-      [[ ... pixel data ... ]]
-   >
+  iex> {:ok, mat} = OpenCV.imread("/path/to/exist/img.png")
+  iex> nx_tensor = OpenCV.Nx.to_nx(mat)
+  ...> #Nx.Tensor<
+  ...>    u8[1080][1920][3]
+  ...>    [[ ... pixel data ... ]]
+  ...> >
   ```
   """
   @doc namespace: :external
   @spec to_mat(reference()) :: {:ok, reference()} | {:error, String.t()}
   def to_nx(mat) do
-    if Code.ensure_loaded?(Nx) do
-      {:ok, mat_type} = OpenCV.Mat.type(mat)
-      {:ok, mat_shape} = OpenCV.Mat.shape(mat)
+    {:ok, mat_type} = OpenCV.Mat.type(mat)
+    {:ok, mat_shape} = OpenCV.Mat.shape(mat)
 
-      case OpenCV.Mat.to_binary(mat) do
-        {:ok, bin} ->
-          bin
-          |> Nx.from_binary(mat_type)
-          |> Nx.reshape(mat_shape)
+    case OpenCV.Mat.to_binary(mat) do
+      {:ok, bin} ->
+        bin
+        |> Nx.from_binary(mat_type)
+        |> Nx.reshape(mat_shape)
 
-        {:error, reason} ->
-          {:error, List.to_string(reason)}
+      {:error, reason} ->
+        {:error, List.to_string(reason)}
 
-        _ ->
-          {:error, "unknown error"}
-      end
-    else
-      {:error, ":nx is missing"}
+      _ ->
+        {:error, "unknown error"}
     end
   end
 
   @doc """
   Converts a tensor of `Nx` to `Mat` of evision (OpenCV).
+
+  If the tensor has three dimensions, it is expected
+  to have shape`{width, height, channels}`.
```

Comment:
I am not sure what happens when you give it something that is not three-dimensional. :)

Reply:
If the tensor is not three-dimensional, then `cv::Mat` will be a general multi-dimensional array. The mat will be initialised by [this function](https://docs.opencv.org/3.4/d3/d63/classcv_1_1Mat.html#a5fafc033e089143062fd31015b5d0f40)

```cpp
Mat(int rows, int cols, int type, void* data, size_t step=AUTO_STEP);
```

And if the tensor is three-dimensional, then we expect the shape to be `{height, width, channels}`. which is the `HWC` format.

Reply:
```suggestion
  to have shape`{height, width, channels}`.
```